### ISSUE TITLE
What's the matter with "clone_fn" ?

### ISSUE THREAD

**Issue by MaeThird** [tensorflow/models#2811]
I'm trying to use "**train_image_classifier.py**" to train model and need to **modify the loss** calculation. But even  small modifications caused the errors. such as:
![t](https://user-images.githubusercontent.com/14851411/32889558-51daf408-cac3-11e7-898f-ba15c48dff6f.jpg) 
got "**Cannot assign a device for operation 'Print': Could not satisfy explicit device specification '/device:GPU:0' because no supported kernel for GPU devices is available**"
such as:
![t3](https://user-images.githubusercontent.com/14851411/32889643-a9537f02-cac3-11e7-9873-f7bb4cbe8ac3.jpg)
got "**Cannot assign a device for operation 'labels_x': Could not satisfy explicit device specification '/device:GPU:0' because no supported kernel for GPU devices is available**"
So Which operations does this deploy-easy module support ?

**Comment by MaeThird** [comment]
`allow_soft_placement=True` can solve.
